src/protocol/sort/generate_sort_permutation.rs

```diff
@@ -147,4 +201,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    pub async fn test_shuffle_and_reveal_permutation() {
+        const BATCHSIZE: u32 = 25;
+
+        let mut rng = rand::thread_rng();
+
+        for _ in 0..10 {
+            let mut permutation: Vec<u32> = (0..BATCHSIZE).collect();
+            permutation.shuffle(&mut rng);
+
+            let world = make_world(QueryId);
+            let [ctx0, ctx1, ctx2] = make_contexts(&world);
+            let permutation: Vec<u128> = permutation.iter().map(|x| u128::from(*x)).collect();
+
+            let [perm0, perm1, perm2] = generate_shares::<Fp31>(&permutation);
+
+            let h0_future = shuffle_and_reveal_permutation(
+                ctx0.narrow("shuffle_reveal"),
+                BATCHSIZE.try_into().unwrap(),
+                perm0,
+            );
+            let h1_future = shuffle_and_reveal_permutation(
+                ctx1.narrow("shuffle_reveal"),
+                BATCHSIZE.try_into().unwrap(),
+                perm1,
+            );
+            let h2_future = shuffle_and_reveal_permutation(
+                ctx2.narrow("shuffle_reveal"),
+                BATCHSIZE.try_into().unwrap(),
+                perm2,
+            );
+
+            let perms_and_randoms: [_; 3] = try_join_all([h0_future, h1_future, h2_future])
```

Comment:
join3() was written for this purpose (the one in the fixture, not the one in the future crate)